### ISSUE TITLE
support sampling of DOIs

### ISSUE THREAD

**Issue by mfenner** [datacite/datacite#317]
As a user of the DataCite REST API, I want to be able to get a sample of the results, so that I do experiments.

### DEV NOTES

* use random dynamic field in Solr
* generate random string in spinone API
* align with [Crossref REST API](https://github.com/CrossRef/rest-api-doc)

The implementation is a `sample` query parameter that takes a number (1-100) for the number of desired results. The `sample` query parameter overwrites the `page[size]` and `sort` query parameters. The other query parameters are not affected.

Example: 10 random DOIs registered by Zenodo:

```
https://api.datacite.org/works?data-center-id=cern.zenodo&sample=10
```

**Comment by mfenner** [comment]
Implemented in `spinone` via datacite/spinone#d578d67b

**Comment by mfenner** [comment]
Related to datacite/freya#17.
